test/coretasks/test_coretasks_sasl.py

```diff
@@ -253,6 +253,26 @@ def test_sasl_plain_no_password(
     ), 'No password is a configuration error and the bot must quit.'
 
 
+# https://github.com/sopel-irc/sopel/issues/2560
+def test_sasl_external_no_password(
+    botfactory: BotFactory,
+    configfactory: ConfigFactory,
+) -> None:
+    tmpconfig = configfactory("conf.ini", TMP_CONFIG_SASL_NO_PASSWORD)
```

Comment:
I'd rather see a new `TMP_CONFIG_SASL_EXTERNAL` constant here, because yes, it doesn't have a password, but it's using a different SASL mechanism, which is more important than "not having a password".